### ISSUE TITLE
problem with installation dependencies on different versions on raspberry pi

### ISSUE THREAD

**Issue by lucasbertolinogb** [Genymobile/scrcpy#5475]
using:
OS: raspbian bullseye
Device model: raspberry pi 1B

Problem: in the first step to use scrcpy my raspberry pi system is not downloading the correct dependencies : 'gcc git pkg-config meson ninja-build libsdl2-dev
         libavcodec-dev libavdevice-dev libavformat-dev libavutil-dev 
         libswresample-dev libusb-1.0-0 libusb-1.0-0-dev'

I use "sudo apt install" before them

But then you would think that it would solve it using "sudo apt --fix-broken install" , but it returns me:
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/libs/libsepol/libsepol1-dev_3.1-1_armf.deb 
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?

At this point I thought "let's see on the website it showed the reason for not downloading"... and guess what? The version that cannot be installed does not exist. Until then, it would be the same as my problem in issue #5423, but I realized that in both versions, both bookworm and bullseye have the same problems installing the dependencies... either I'm downloading the error or the system is making me download the wrong dependencies. ....

**Comment by rom1v** [comment]
> E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?

Did you run `sudo apt update`?

**Comment by lucasbertolinogb** [comment]
Yes, on my Raspberry it only works with the use of sudo because the internet I'm using comes from a college and there is a block for some commands like install, but commands like npm work normally

**Comment by lucasbertolinogb** [comment]
Ohhh wait... I forgot to say that in the problem it is also written at the end of the first error line 
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/libs/libsepol/libsepol1-dev_3.1-1_armf.deb 404 Not Found [IP: xx.xx.xxx.xxx xxx]

**Comment by lucasbertolinogb** [comment]
> > E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
> 
> Did you run `sudo apt update`?

I run sudo apt update and it returns me:
E: Release file for http://raspbian.raspberrypi.org/raspbian/dists/bullseye/InRelease is not valid yet (invalid for another 20d 12h 21min 14s). Updates for this repository will not be applied.

**Comment by rom1v** [comment]
Is your system date correct?

**Comment by lucasbertolinogb** [comment]
<.< ... he says this in: Tue 22 Oct 2024 04:19:14 PM UTC

**Comment by lucasbertolinogb** [comment]
> Is your system date correct?

<.< ... he says this in: Tue 22 Oct 2024 04:19:14 PM UTC -03 ... how can i adjust the date in linux ?

**Comment by lucasbertolinogb** [comment]
latest update on this issue:
After mounths I decided to buy a Raspberry Pi 5 because the Raspberry I used is like Pi 1 or something similar and there were a lot of slowdown problems (I would never think that opening a folder could take more than 10 seconds :p)... after receiving this new hardware I did a small test using timedatectl:

sudo timedatectl set-ntp false 
sudo timedatectl set-time 'year-month-day hours:minutes:seconds'
ex: sudo timedatectl set-time '2025-02-05 8:50:00'
sudo timedatectl set-ntp true

After these instructions I copied the standard scrcpy scripts and... it worked (I couldn't believe it) so the problem really was that I lacked the money to buy a newer one :D, but the only problem it generates, which at least is easy to solve, would be that every time I download something I would have to follow the same previous commands, but it could be solved by buying an RTC for the raspberry
